pyvips/vimage.py

```diff
@@ -998,10 +998,17 @@ def __getitem__(self, arg):
             last_two = rgb_image[1:]
             last_band = rgb_image[-1]
             middle_few = multiband[1:-2]
+            reversed  = multiband[::-1]
+            every_other = multiband[::2]
+            other_every_other = multiband[1::2]
 
         """
 
         if isinstance(arg, slice):
+            if arg.step is not None: # fancy slicing, though this would work for any slice
+                bands = [ self.extract_band(i) for i in range(self.bands)[arg] ]
+                return bands[0].bandjoin(bands[1:])
+
```

Comment:
I try to avoid early returns if possible, except for error checks.

Reply:
Latest implementation addresses this.